src/components/UI/ErrorFallback.js

```diff
@@ -11,22 +14,43 @@ const ErrorWrapperSpan = styled.span`
   font-weight: 300;
 `;
 
-// This function joins error messages into a single string if it receives an array,
-// or returns the error if it is a single value of type string or null.
-// If an array of null values are passed, it returns null.
-function parseErrors(errors) {
+/**
+ * Helper that finds the best way to return a pretty-printed
+ * version of the error object passed to it
+ * @param {?(string[]|string)} errors - Error message(s)
+ * @param {string} [separator=". "] - Separator between messages (if passing an error list)
+ * @return {?string}
+ */
+function parseErrors(errors, separator = '. ') {
   if (typeof errors === 'string' || !errors) return errors;
 
-  // Remove null values and joins values in a string
-  // TODO maybe we want to show errors in a different manner.
-  const notNullErrors = errors.filter(error => error);
-  if (notNullErrors.length === 0) return null;
+  let message = DEFAULT_MESSAGE;
+
+  if (Array.isArray(errors)) {
```

Comment:
Kept the check here because we'd get problems again if we want to use the `ErrorFallback` component with actions that already dispatch error objects